### ISSUE TITLE
rule idea: recursive directory listing

### ISSUE THREAD

**Issue by williballenthin** [fireeye/capa-rules#26]

**Comment by re-fox** [comment]
I've been looking through my corpus to find more samples that target directories. 

An easy win may be something like

```yaml
rule:
  meta:
    name: enumerate files recursively
    namespace: host-interaction/file-system/files/list
    author: "@_re_fox"
    scope: function
    att&ck:
      - Discovery::File and Directory Discovery [T1083]
    examples:
      - 5f66b82558ca92e54e77f216ef4c066c:0x40640E
  features:
    - and:
      - or:
        - match: enumerate files via kernel32 functions
        - match: enumerate files via ntdll functions
      - characteristic: recursive call
```

This rule doesn't target directories, but if we wanted to be more explicit, we could use `0x10 = FILE_ATTRIBUTE_DIRECTORY` (gathered from here-> https://docs.microsoft.com/en-us/windows/win32/fileio/file-attribute-constants) which is used to determine folders in code like 
```c
    if (FileAttributes & FILE_ATTRIBUTE_DIRECTORY) {
        printf("Directory ");
    }
```

**Comment by williballenthin** [comment]
this looks like a good place to start.

**Comment by williballenthin** [comment]
closed in #195
